pkg/cap/cluster/anchor/anchor.go

```diff
@@ -0,0 +1,34 @@
+package anchor
+
+import (
+	"context"
+	"errors"
+)
+
+var (
+	ErrInvalidPath = errors.New("invalid path")
+)
+
+type Anchor interface {
+	Path() []string
+	Ls(ctx context.Context) (AnchorIterator, error)
+	Walk(ctx context.Context, path []string) (Anchor, error)
+}
+
+type AnchorIterator interface {
+	Next(context.Context) bool
+	Finish()
+	Anchor() Anchor
+	Err() error
+}
+
+type Host interface {
+	Anchor
+	Host() string
+}
+
+type Container interface {
+	Anchor
```

Comment:
>  I'm not convinced we need a release function, here. If the returned type needs to be released, it should probably have a .Release method. Either way, do we need it right now?

If my understanding of capnproto is right, when we receive a future we need to release it at some point right? Otherwise, the server will have an extra reference of the capability, right? When we do a `Walk` or `Ls` call we obtain a capability future, and when we end up using it we should therefore call release on it, don't we?

Reply:
Ah yes ok.  In that case, I think we would need to return `(Future, func())`.  The current method signature implies a call to `f.Struct()` under the hood, which blocks until the promise is resolved, at which point we can call `release` before returning `([]byte, error)`.

This is all detail, though.  Let's not worry about it yet.  Feel free to leave this as-is for the time being.